x-pack/plugins/security_solution/server/endpoint/services/artifacts/manifest_manager/manifest_manager.ts

```diff
@@ -592,81 +622,93 @@ export class ManifestManager {
-          const serializedManifest = manifest.toPackagePolicyManifest(id);
-
-          if (!manifestDispatchSchema.is(serializedManifest)) {
-            errors.push(new EndpointError(`Invalid manifest for policy ${id}`, serializedManifest));
-          } else if (!manifestsEqual(serializedManifest, oldManifest.value)) {
-            packagePolicy.inputs[0].config.artifact_manifest = { value: serializedManifest };
-            packagePoliciesToUpdate.push(packagePolicy);
+        if (response.updatedPolicies) {
+          updatedPolicies.push(
+            ...response.updatedPolicies.map((policy) => {
+              return `[${policy.id}][${policy.name}] updated with manifest version: [${manifestVersion}]`;
+            })
+          );
+        }
+      },
+    });
+
+    for await (const policies of this.fetchAllPolicies()) {
+      for (const packagePolicy of policies) {
+        const { id, name } = packagePolicy;
+
+        if (packagePolicy.inputs.length > 0 && packagePolicy.inputs[0].config !== undefined) {
+          const oldManifest = packagePolicy.inputs[0].config.artifact_manifest ?? {
+            value: {},
+          };
+
+          const newManifestVersion = manifest.getSemanticVersion();
+
+          if (semver.gt(newManifestVersion, oldManifest.value.manifest_version)) {
+            const serializedManifest = manifest.toPackagePolicyManifest(id);
+
+            if (!manifestDispatchSchema.is(serializedManifest)) {
+              errors.push(
+                new EndpointError(`Invalid manifest for policy ${id}`, serializedManifest)
+              );
+            } else if (!manifestsEqual(serializedManifest, oldManifest.value)) {
+              packagePolicy.inputs[0].config.artifact_manifest = { value: serializedManifest };
+              policyUpdateBatchProcessor.addToQueue(packagePolicy);
+            } else {
+              unChangedPolicies.push(`[${id}][${name}] No change in manifest content`);
+            }
           } else {
-            this.logger.debug(
-              `No change in manifest content for package policy: ${id}. Staying on old version`
-            );
+            unChangedPolicies.push(`[${id}][${name}] No change in manifest version`);
           }
         } else {
-          this.logger.debug(`No change in manifest version for package policy: ${id}`);
+          errors.push(
+            new EndpointError(`Package Policy ${id} has no 'inputs[0].config'`, packagePolicy)
+          );
         }
-      } else {
-        errors.push(
-          new EndpointError(`Package Policy ${id} has no 'inputs[0].config'`, packagePolicy)
-        );
       }
-    });
+    }
 
-    // Split updates in batches with batch size: packagerTaskPackagePolicyUpdateBatchSize
-    const updateBatches = chunk(
-      packagePoliciesToUpdate,
-      this.packagerTaskPackagePolicyUpdateBatchSize
+    await policyUpdateBatchProcessor.complete();
+
+    this.logger.info(
```

Comment:
should this be a `debug` and only display info logs if there are policy changes (as done below)?

Reply:
same reason here - I wanted the task to output what it actually did in a brief way